test/e2e/postinstall/syncsets/syncsets_suite_test.go

```diff
@@ -0,0 +1,517 @@
+										},
+										ObjectMeta: metav1.ObjectMeta{
+											Name:      "foo-selectorsyncset",
+											Namespace: "default",
+										},
+										Data: map[string]string{
+											"foo-sSS": "bar-sSS",
+										},
+									},
+								},
+							},
+							Patches: []hivev1.SyncObjectPatch{
+								{
+									APIVersion: "v1",
+									Kind:       "ConfigMap",
+									Name:       "foo",
+									Namespace:  "default",
+									Patch:      `{ "data": { "foo": "new-bar" } }`,
+									PatchType:  "merge",
+								},
+							},
+							Secrets: []hivev1.SecretMapping{
+								{
+									SourceRef: hivev1.SecretReference{
+										Name:      pullSecret,
+										Namespace: clusterNamespace,
+									},
+									TargetRef: hivev1.SecretReference{
+										Name:      "test-pull-secret-copy",
+										Namespace: "default",
+									},
+								},
+							},
+						},
+					},
+				}
+
+				By("Creating a ConfigMap on target cluster")
+				err = targetClusterClient.Create(ctx, resourceConfigMap)
+				Ω(err).ShouldNot(HaveOccurred())
+				time.Sleep(5 * time.Second)
+
+				By("Verify resource ConfigMap created on target cluster")
+				resultConfigMap := &corev1.ConfigMap{}
+				err = targetClusterClient.Get(ctx, client.ObjectKey{Name: "foo", Namespace: "default"}, resultConfigMap)
+				Ω(err).ShouldNot(HaveOccurred())
+				Ω(resultConfigMap.Data).Should(Equal(map[string]string{"foo": "bar"}))
+
+				By("Creating a selectorSyncSet on target cluster including resources, patches and secretMappings")
+				err = hiveClient.Create(ctx, SelectorSyncSet)
+
+				err = waitForSyncSetInstanceApplied("test-selectorsyncset", "selectorsyncset", clusterNamespace)
+				Ω(err).ShouldNot(HaveOccurred())
+
+				By("Verifying the resource was synced")
+				resultConfigMap = &corev1.ConfigMap{}
+				err = targetClusterClient.Get(ctx, client.ObjectKey{Name: "foo-selectorsyncset", Namespace: "default"}, resultConfigMap)
+				Ω(err).ShouldNot(HaveOccurred())
+				Ω(resultConfigMap.Data).Should(Equal(map[string]string{"foo-sSS": "bar-sSS"}))
+
+				By("Verifying the resource was patched")
+				resultConfigMap = &corev1.ConfigMap{}
+				err = targetClusterClient.Get(ctx, client.ObjectKey{Name: "foo", Namespace: "default"}, resultConfigMap)
+				Ω(err).ShouldNot(HaveOccurred())
+				Ω(resultConfigMap.Data).Should(Equal(map[string]string{"foo": "new-bar"}))
+
+				By("Verifying the secret was copied")
+				resultSecret := &corev1.Secret{}
+				err = targetClusterClient.Get(ctx, client.ObjectKey{Name: "test-pull-secret-copy", Namespace: "default"}, resultSecret)
+				Ω(err).ShouldNot(HaveOccurred())
+
+				By(`Deleting the label "cluster-group: hivecluster" of clusterdeployment`)
+				cdLabels = cd.ObjectMeta.GetLabels()
+				delete(cdLabels, "cluster-group")
+				cd.ObjectMeta.SetLabels(cdLabels)
+				hiveClient.Update(ctx, cd)
+				Ω(cd.ObjectMeta.Labels).Should(Equal(cdLabels))
+				time.Sleep(5 * time.Second)
```

Comment:
Instead of sleeping, you could wait for the syncsetinstance to be deleted, using `clientwatch.UntilWithSync`